File: backend/app/services/storage/none.py
"""none StorageProvider：不转存，直接使用 Agnes 原始 URL。"""
from __future__ import annotations

from typing import Optional

from app.services.storage.base import StorageProvider, StorageUploadResult


class NoneProvider(StorageProvider):
    name = "none"

    def upload_bytes(
        self,
        data: bytes,
        media_type: str = "img",
        ext: str = "png",
        content_type: Optional[str] = None,
    ) -> Optional[StorageUploadResult]:
        return None

    def upload_from_url(
        self,
        source_url: str,
        media_type: str = "img",
        ext: Optional[str] = None,
        content_type: Optional[str] = None,
    ) -> Optional[StorageUploadResult]:
        return None

    def get_display_url(self, key: str) -> str:
        return ""

File: packages/provider/src/storage/none.ts
import type { ObjectStorage, StorageUploadInput, StorageUploadResult } from '../object-storage.interface.js';

/**
 * none 对象存储：不转存，返回 null，由调用方降级保留上游原始 URL。
 */
export class NoneObjectStorage implements ObjectStorage {
  readonly provider = 'none';

  async uploadBytes(_data: Buffer, _input?: StorageUploadInput): Promise<StorageUploadResult | null> {
    return null;
  }

  async uploadFromUrl(_sourceUrl: string, _input?: StorageUploadInput): Promise<StorageUploadResult | null> {
    return null;
  }

  async uploadFile(_filePath: string, _input?: StorageUploadInput): Promise<StorageUploadResult | null> {
    return null;
  }

  async getDisplayUrl(_key: string): Promise<string> {
    return '';
  }
}